addons/sale_project/models/sale_order.py

```diff
@@ -69,7 +69,9 @@ def action_view_task(self):
             if action.get('context'):
                 eval_context = self.env['ir.actions.actions']._get_eval_context()
                 eval_context.update({'active_id': task_projects.id})
-                action['context'] = safe_eval(action['context'], eval_context)
+                action_context = safe_eval(action['context'], eval_context)
+                action_context.update(eval_context)
+                action['context'] = action_context
```

Comment:
Are you sure that this modification fix the issue?

Reply:
Yes, the active_id wasn't being added to the context so the wrong informations were being displayed.